frontend/src/lib/k8s/KubeObject.ts

```diff
@@ -0,0 +1,561 @@
+import { OpPatch } from 'json-patch';
+import { JSONPath } from 'jsonpath-plus';
+import { cloneDeep, unset } from 'lodash';
+import React from 'react';
+import helpers from '../../helpers';
+import { getCluster } from '../cluster';
+import { createRouteURL } from '../router';
+import { timeAgo } from '../util';
+import { useConnectApi } from '.';
+import { useKubeObject, useKubeObjectList } from './api/v2/hooks';
+import { ApiError, apiFactory, apiFactoryWithNamespace, post, QueryParameters } from './apiProxy';
+import {
+  ApiListOptions,
+  ApiListSingleNamespaceOptions,
+  AuthRequestResourceAttrs,
+  KubeMetadata,
+  KubeObjectClass,
+  KubeObjectInterface,
+} from './cluster';
+import { KubeEvent } from './event';
+
+function getAllowedNamespaces() {
+  const cluster = getCluster();
+  if (!cluster) {
+    return [];
+  }
+
+  const clusterSettings = helpers.loadClusterSettings(cluster);
+  return clusterSettings.allowedNamespaces || [];
+}
+
+export class KubeObject<T extends KubeObjectInterface | KubeEvent = any> {
+  jsonData: T;
+  /** Readonly field defined as JSONPath paths */
+  static readOnlyFields: string[] = [];
+  readonly _clusterName: string;
+  cluster?: string;
```

Comment:
I think these should be the same, right? And we use resource.cluster in the recent multicluster changes, so this `cluster` var needs to be assigned to the cluster name in the constructor or we need to make this a getter (and just return the _clusterName) which is likely easier.

Reply:
yep you're right